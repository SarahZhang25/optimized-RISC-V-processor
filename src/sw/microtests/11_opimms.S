// This is a test that tests that your processor handles all the OPIMM
// instructions correctly.
.text
start:
    li x1, 0x123
    addi x2, x1, -0x321
    andi x3, x2, -0xab
    ori  x4, x3, 0xf0
    xori x5, x4, 0xcd
    slli x6, x5, 3
    srai x7, x6, 4
    srli x8, x7, 5

    slti  x9 , x7, 0
    sltiu x10, x7, 0
    slti  x11, x7, -1
    sltiu x12, x7, -1
    slti  x13, x7, -0x800
    sltiu x14, x7, -0x800
    slti  x15, x7, 0x7ff
    sltiu x16, x7, 0x7ff

    slti  x17, x10, 0
    sltiu x18, x10, 0
    slti  x19, x10, -1
    sltiu x20, x10, -1
    slti  x21, x10, -0x800
    sltiu x22, x10, -0x800
    slti  x23, x10, 0x7ff
    sltiu x24, x10, 0x7ff
