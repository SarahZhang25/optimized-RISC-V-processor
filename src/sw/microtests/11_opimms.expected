Dumping the state of the processor
pc = 0x00000060
x0 = 0x00000000
x1 = 0x00000123
x2 = 0xfffffe02
x3 = 0xfffffe00
x4 = 0xfffffef0
x5 = 0xfffffe3d
x6 = 0xfffff1e8
x7 = 0xffffff1e
x8 = 0x07fffff8
x9 = 0x00000001
x10 = 0x00000000
x11 = 0x00000001
x12 = 0x00000001
x13 = 0x00000000
x14 = 0x00000000
x15 = 0x00000001
x16 = 0x00000000
x17 = 0x00000000
x18 = 0x00000000
x19 = 0x00000000
x20 = 0x00000001
x21 = 0x00000000
x22 = 0x00000001
x23 = 0x00000001
x24 = 0x00000001
x25 = 0x00000000
x26 = 0x00000000
x27 = 0x00000000
x28 = 0x00000000
x29 = 0x00000000
x30 = 0x00000000
x31 = 0x00000000
Quitting simulation.
